src/messaging/helpers/email-envelope.js

```diff
@@ -0,0 +1,169 @@
+import Utils from '../../utils'
+
+function checkCriteria(criteria) {
+  if (criteria) {
+    throw new Error('Addresses can not be set if criteria is already set in a builder')
+  }
+}
+
+export default class EmailEnvelop {
+
+  static create() {
+    return new EmailEnvelop()
+  }
+
+  constructor() {
+    this.addresses = []
+    this.ccAddresses = []
+    this.bccAddresses = []
+    this.criteria = null
+  }
+
+  /**
+   *
+   * @param {Array|String} addresses
+   * @returns {EmailEnvelop}
+   */
+  setTo(addresses) {
```

Comment:
all the methods must be described in `backendless.d.ts`, and JSDoc, I guess, will be redundant here